### ISSUE TITLE
Create support admin page to allow of deletion of facility

### ISSUE THREAD

**Issue by BobanL** [CDCgov/prime-simplereport#5847]
### Background
To empower the support team to resolve user issues more quickly, we should provide them functionality to delete facilities in an organization using the existing mutation to markFacilitiesAsDeleted. 

### Action requested
Create a page only accessible to support admins to manage facilities. The page should have the ability to select an organization, a facility from that organization, and buttons to either delete or undelete the facility. 
- Backend mutation already exists to handle the deletion/undeletion of a facility. 
- To fetch all organizations & facilities the organizations query will likely have to be used. 
Rough example:
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/628b9f2d-5467-446f-968a-e4a020f4a1cc)

### Acceptance Criteria
- The page is accessible and passes AXE scans. 
- A user is able to delete and undelete a facility.

**Comment by mehansen** [comment]
Open questions:
* what happens to pxp flow if patient viewing a result performed at a deleted facility
* what happens to patient assigned to a facility that gets deleted - what do we see in Edit Patient and can we conduct a test for them from anywhere?
  * should we warn support before deleting a facility if there are patients only assigned to that facility because they will potentially become untestable?
* does deleting a facility impact the result view table?
* what happens if a user logs in and they have access to no facilities?

creating a spike to answer these first:
proposes a solution to preventing getting into any bad states when deleting a facility (either by preventing deletion if users assigned to only that facility or making support remove all users from facility before deletion.. also still need to check what happens to results for a deleted facility)

**Comment by mehansen** [comment]
@jayna-SkylightDigital please add expected behavior when:
- patient in testing queue of facility to be deleted
- there are patients assigned to the facility (would become inaccessible if facility is deleted)
- users assigned to only that facility
  - confirm if we're okay with them seeing the warning message that they don't have access to any facilities

see #5977 for notes on corner cases

**Comment by mehansen** [comment]
blocked by #6115

**Comment by mehansen** [comment]
should we support self-service undeleting of facilities? we didn't do much spike work around this and it's not a common support request (but does seem like it would be easy to add)

**Comment by jayna-SkylightDigital** [comment]
### Expected behavior:

**Patients in Testing Queue of Facility to be Deleted**

- If a SimpleReport user attempts to delete a facility that has active tests in the testing queue, a warning message should be displayed to inform the user that if the facility deletion is completed, the patient's test will be cancelled and removed from the queue. 
- The user may either cancel out of the deletion process to address the tests currently in the queue, or proceed with the deletion and confirm their acknowledgement that any tests currently in the queue will be deleted.

**Patients Assigned to the Facility**

- A SimpleReport user may not delete a facility if there are patients still assigned to that facility. 
- If a SimpleReport user attempts to delete a facility that still has patients assigned to that facility, a warning message should be displayed informing the user that the patients in that facility must be reassigned to a different facility before the deletion process can proceed. 

Open questions: 

- Can the system allow for bulk reassignment of patients to another facility with a single click? 
- If the organization has no other facilities, should the warning message inform the user that a new facility must be created before patients can be reassigned?

**Users Assigned to Only the Facility Being Deleted**

- If a SimpleReport user was assigned to a facility that has been deleted (and is not assigned to any other facilities), they will still be able to log into SimpleReport, but they will lose access to all other SimpleReport functionality until they are assigned to a new facility by their organization administrator.
- Upon login, SimpleReport should display a warning message to these users informing them that their assigned facility has been deleted, and they must contact their organization administrator to be reassigned to a different facility.
- A SimpleReport user should not be able to delete their own facility if they are not assigned to any other facilities.

Open questions:
- is there a way to make it easy for this user to reach out to their administrator? (provide email on screen, or one-click to contact via email)

**Comment by mehansen** [comment]
@jayna-SkylightDigital to create a follow-up ticket with nice-to-haves from [this comment](https://github.com/orgs/CDCgov/projects/15/views/1?pane=issue&itemId=26603532)

for now, we'll just do a warning modal (see updated acceptance criteria)
